include/rmm/device_buffer.hpp

```diff
@@ -202,6 +201,7 @@ class device_buffer {
    * replaced by the `other.stream()`.
    *
    * @param other The `device_buffer` whose contents will be moved.
+   * @return device_buffer& A reference to this `device_buffer`.
```

Comment:
I think the documentation guide says to not put the return type in the @return docs. It's obvious and Doxygen generates it.

```suggestion
   * @return A reference to this `device_buffer`.
```